src/nvim/api/buffer.c

```diff
@@ -514,24 +514,25 @@ void nvim_buf_set_lines(uint64_t channel_id, Buffer buffer, Integer start, Integ
 
 /// Sets (replaces) a range in the buffer
 ///
-/// This is recommended over nvim_buf_set_lines when only modifying parts of a
-/// line, as extmarks will be preserved on non-modified parts of the touched
+/// This is recommended over |nvim_buf_set_lines()| when only modifying parts of
+/// a line, as extmarks will be preserved on non-modified parts of the touched
 /// lines.
 ///
-/// Indexing is zero-based and end-exclusive.
+/// Row indices are zero-based and end-inclusive. Column indices are zero-based,
+/// end-exclusive, and correspond to bytes, not display columns.
```

Comment:
Why mention "display columns"? There are many column types, and mentioning one and only one of them is strange.

Reply:
The reason I specifically mentioned it is that I assumed as a user of this API that the indices were display columns -- since the arguments are called "columns". What are the other column types?

Reply:
At least there are character column and virtual column.

Reply:
I am 100% open to revisiting the terminology and making it consistent with column naming elsewhere in the docs, I agree that's important. Do you object to any usage of the phrasing "...bytes, not <some kind of> columns"? Because I do think as a user that something like this is helpful.

Reply:
I guess the real reason I picked "display columns" is that in my mind "display column = character column". What about "...bytes, not character columns"? I don't think anyone would ever assume this means "virtual columns".

Reply:
In most places "column" means byte column. There is an example in `:h getpos()`:
```
		The column number in the returned List is the byte position
		within the line. To get the character position in the line,
		use |getcharpos()|.
```

Reply:
First, I think that given the number of types of columns there are, it would have been very bad to introduce a new term "display column" (which also does not appear anywhere else in the codebase). I probably shouldn't have used the term without more due diligence. Anyway thanks for commenting on this line.

I've made a change in this paragraph in both places to use the term "character column", and additionally, I added references to `col()` and `charcol()`, which describe what these two column types actually are. My rationale is, this function `vim.api.nvim_buf_set_text` is very likely to be the first place that a new plugin author or hacker will land when editing text programmatically, especially someone not previously familiar with the Vim script interfaces -- speaking as someone who just went through this experience. The docs for `col()` and `charcol()` are never going away or changing, so there is no maintenance risk to linking them here. I think that in general it makes sense to be judicious including cross-refs in docs, but IMO it's very worth having something in the docs for `vim.api.nvim_buf_[get|set]_text` that points the user at a description of the difference between byte columns and character columns.

Reply:
Will revisit later.